EducationPortal/EducationPortal.ConsoleUI/Validation/CourseDataValidator.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using EducationPortal.BLL;
+using EducationPortal.BLL.DTO;
+
+namespace EducationPortal.ConsoleUI.Validation
+{
+    public class CourseDataValidator
+    {
+        private CourseDTO course;
+
+        public CourseDataValidator(CourseDTO course)
+        {
+            this.course = course;
+        }
+
+        public ValidationResult Validate()
+        {
+            var result = ValidateName();
+
+            if (!result.IsValid)
+            {
+                return result;
+            }
+
+            result = ValidateDescription();
+
+            if (!result.IsValid)
```

Comment:
You are overriding the result instance

Reply:
there's no need to create few instances, because if previous validation was successful, "result" can be reused for the next one